Add logs section

nmuesch opened the pull request:
### What does this PR do?

Adds logs section to readme and example yaml file. 
Adds the log collection category to the manifest file

Most of this info came from here - http://www.postfix.org/BASIC_CONFIGURATION_README.html#syslog_howto

### Motivation

We now have an OOTB logs pipeline for postfix so update the docs and yaml to reflect that. 

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If PR adds a configuration option, it has been added to the configuration file.

### Additional Notes

Anything else we should know when reviewing?

phrawzty commented on postfix/README.md:
```diff
@@ -78,6 +78,41 @@ List of users who are authorized to view the queue.
+
+**Available for Agent >6.0**
+
+Postfix logs activity to the syslog daemon that then gets sent to logfiles.
```
```suggestion
Postfix sends logs to the syslog daemon, which then writes logs to the file system.
```

phrawzty commented on postfix/README.md:
```diff
@@ -78,6 +78,42 @@ List of users who are authorized to view the queue.
+
+Postfix logs activity to the syslog daemon that then gets sent to logfiles.
+
+By default, the naming convention and log files can by configured in your syslog file:
```
```suggestion
The naming convention and log file destinations are configurable:
```

phrawzty commented on postfix/README.md:
````diff
@@ -78,6 +78,42 @@ List of users who are authorized to view the queue.
+```
+
+
+* Collecting logs is disabled by default in the Datadog Agent, enable it in your `datadog.yaml` file with:
````
```suggestion
* Collecting logs is disabled by default in the Datadog Agent. Enable it in your `datadog.yaml` file:
```

nmuesch replied:
This is present in a few other README's as well, as I took it from there. But I can change it to this :)

l0k0ms replied:
The double sentences one is better since it's more "business". Simple sentence for simple concepts.

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## Log Section (Available for Agent >=6.0)
+##
+## type - mandatory - Type of log input source (tcp / udp / file / windows_event)
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
```
> Set path if type is file and channel if windows_event

That is super confusing. Can you elaborate?

l0k0ms replied:
it's described in the log documentation, but depending of the type, the next parameter changes

groups of configuration are: 

```
 - type: file
    path: /EZF/EZF/EF
```

```
   - type: tcp
      port: 1234
```

```
   - type: udp
      port: 1234
```

```
   - type: windows_evnet
      path: /ZEA/ZEAR/ZAER
```

phrawzty replied:
```suggestion
## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file. Set channel if type is windows_event
```

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+##
+## type - mandatory - Type of log input source (tcp / udp / file / windows_event)
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
+## service - mandatory - Name of the service owning the log
```
```suggestion
## service - mandatory - Name of the service that owns the log
```

Is that accurate? Or is it the name of the service that generates the logs, for example?

l0k0ms replied:
it's the name of the service that generates the logs not own it, since we want to bind those logs to the corresponding APM service.

phrawzty replied:
Ok, got it. Let's be clear about that.
```suggestion
## service - mandatory - Name of the service that generated the log
```

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## type - mandatory - Type of log input source (tcp / udp / file / windows_event)
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
+## service - mandatory - Name of the service owning the log
+## source  - mandatory - Attribute that defines which integration is sending the logs
```
```suggestion
## source  - mandatory - Attribute that defines which Integration is sending the logs
```

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
+## service - mandatory - Name of the service owning the log
+## source  - mandatory - Attribute that defines which integration is sending the logs
+## sourcecategory - optional - Multiple value attribute. Can be used to refine the source attribtue
```
"Can be used", implying that it this attr can do other things, too. What other things?

l0k0ms replied:
right now it's used just for this.

NBParis replied:
Feel free to remove it, as we plan to change the behaviour for this and it seems to not bring much clarity anywhere.

nmuesch replied:
Removed

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## service - mandatory - Name of the service owning the log
+## source  - mandatory - Attribute that defines which integration is sending the logs
+## sourcecategory - optional - Multiple value attribute. Can be used to refine the source attribtue
+## tags: - optional - Add tags to each logs collected
```
```suggestion
## tags: - optional - Add tags to the collected logs
```

l0k0ms replied:
this can be changed.

nmuesch commented:
Thanks @l0k0ms and @phrawzty ! I've addressed the reviews above. I'll keep this PR scoped to just the postfix integration, but we can work on fixing the other integration's configs/readme-s for clarity in a future PR.

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## type - mandatory - Type of log input source (tcp / udp / file / windows_event)
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
+## service - mandatory - Name of the service owning the log
+## source  - mandatory - Attribute that defines which Integration is sending the logs
```
```suggestion
## source  - mandatory - Attribute that defines which Integration sent the logs
```

phrawzty commented on postfix/datadog_checks/postfix/data/conf.yaml.example:
```diff
@@ -57,3 +57,21 @@ instances:
+## port / path / channel - mandatory - Set port if type is tcp or udp. Set path if type is file and channel if windows_event
+## service - mandatory - Name of the service owning the log
+## source  - mandatory - Attribute that defines which Integration is sending the logs
+## sourcecategory - optional - Multiple value attribute. Can be used to refine the source attribtue
```
```suggestion
## sourcecategory - optional - Multiple value attribute. Used to refine the source attribute
```

nmuesch commented:
Thanks for the suggestions/review. I think I've addressed everything @phrawzty Could you take another look?
